packages/uspf/uspf.0.0.1/opam

```diff
@@ -0,0 +1,41 @@
+opam-version: "2.0"
+maintainer:   "Romain Calascibetta <romain.calascibetta@gmail.com>"
+authors:      "Romain Calascibetta <romain.calascibetta@gmail.com>"
+homepage:     "https://github.com/dinosaure/uspf"
+bug-reports:  "https://github.com/dinosaure/uspf/issues"
+dev-repo:     "git+https://github.com/dinosaure/uspf.git"
+doc:          "https://dinosaure.github.io/uspf/"
+license:      "MIT"
+synopsis:     "SPF implementation in OCaml"
+description: """uspf is an implementation of the SPF verifier
+in OCaml compatible with MirageOS."""
+
+build: [ "dune" "build" "-p" name "-j" jobs ]
+run-test: [ "dune" "runtest" "-p" name "-j" jobs ]
+
+depends: [
+  "ocaml"       {>= "4.07.0"}
+  "dune"        {>= "2.8.0"}
+  "logs"
+  "colombe"
+  "mrmime"      {>= "0.5.0"}
+  "ipaddr"      {>= "5.2.0"}
+  "hmap"
+  "angstrom"    {>= "0.15.0"}
+  "domain-name"
+  "dns"         {>= "5.0.1"}
+  "lwt"
+  "dns-client"  {>= "6.1.0"}
+  "astring"
+  "alcotest"    {with-test}
+  "rresult"     {with-test}
+]
```

Comment:
```suggestion
]

conflicts: [
  "result" {< "1.5"}
]
```

Reply:
I think this is to fix
```

#=== ERROR while compiling uspf.0.0.1 =========================================#
# context              2.1.1 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/dinosaure/uspf/releases/download/0.0.1/uspf-0.0.1.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/uspf.0.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p uspf -j 31
# exit-code            1
# env-file             ~/.opam/log/uspf-16415-ce1b27.env
# output-file          ~/.opam/log/uspf-16415-ce1b27.out
### output ###
#       ocamlc lib/.uspf.objs/byte/uspf.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.uspf.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base64/rfc2045 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigarray-overlap -I /home/opam/.opam/4.08/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/coin -I /home/opam/.opam/4.08/lib/colombe -I /home/opam/.opam/4.08/lib/colombe/emile -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/emile -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/hmap -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ke -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mrmime -I /home/opam/.opam/4.08/lib/pecu -I /home/opam/.opam/4.08/lib/prettym -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rosetta -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/unstrctrd -I /home/opam/.opam/4.08/lib/unstrctrd/parser -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/uuuu -I /home/opam/.opam/4.08/lib/yuscii -intf-suffix .ml -no-alias-deps -open Uspf__ -o lib/.uspf.objs/byte/uspf.cmo -c -impl lib/uspf.ml)
# File "lib/uspf.ml", line 683, characters 13-33:
# 683 |           @@ R.failwith_error_msg (Macro.expand_macro ctx macro)) in
#                    ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound value R.failwith_error_msg
#     ocamlopt lib/.uspf.objs/native/uspf.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I lib/.uspf.objs/byte -I lib/.uspf.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base64/rfc2045 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigarray-overlap -I /home/opam/.opam/4.08/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/coin -I /home/opam/.opam/4.08/lib/colombe -I /home/opam/.opam/4.08/lib/colombe/emile -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/emile -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/hmap -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ke -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mrmime -I /home/opam/.opam/4.08/lib/pecu -I /home/opam/.opam/4.08/lib/prettym -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rosetta -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/unstrctrd -I /home/opam/.opam/4.08/lib/unstrctrd/parser -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/uuuu -I /home/opam/.opam/4.08/lib/yuscii -intf-suffix .ml -no-alias-deps -open Uspf__ -o lib/.uspf.objs/native/uspf.cmx -c -impl lib/uspf.ml)
# File "lib/uspf.ml", line 683, characters 13-33:
# 683 |           @@ R.failwith_error_msg (Macro.expand_macro ctx macro)) in
#                    ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound value R.failwith_error_msg

-       ocamlc lib/.uspf.objs/byte/uspf.{cmo,cmt} (exit 2)
- (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.uspf.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base64/rfc2045 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigarray-overlap -I /home/opam/.opam/4.08/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/coin -I /home/opam/.opam/4.08/lib/colombe -I /home/opam/.opam/4.08/lib/colombe/emile -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/emile -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/hmap -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ke -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mrmime -I /home/opam/.opam/4.08/lib/pecu -I /home/opam/.opam/4.08/lib/prettym -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rosetta -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/unstrctrd -I /home/opam/.opam/4.08/lib/unstrctrd/parser -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/uuuu -I /home/opam/.opam/4.08/lib/yuscii -intf-suffix .ml -no-alias-deps -open Uspf__ -o lib/.uspf.objs/byte/uspf.cmo -c -impl lib/uspf.ml)
- File "lib/uspf.ml", line 683, characters 13-33:
- 683 |           @@ R.failwith_error_msg (Macro.expand_macro ctx macro)) in
-                    ^^^^^^^^^^^^^^^^^^^^
- Error: Unbound value R.failwith_error_msg
-     ocamlopt lib/.uspf.objs/native/uspf.{cmx,o} (exit 2)
- (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I lib/.uspf.objs/byte -I lib/.uspf.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base64/rfc2045 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigarray-overlap -I /home/opam/.opam/4.08/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/coin -I /home/opam/.opam/4.08/lib/colombe -I /home/opam/.opam/4.08/lib/colombe/emile -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/emile -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/hmap -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ke -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mrmime -I /home/opam/.opam/4.08/lib/pecu -I /home/opam/.opam/4.08/lib/prettym -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rosetta -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/unstrctrd -I /home/opam/.opam/4.08/lib/unstrctrd/parser -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/uuuu -I /home/opam/.opam/4.08/lib/yuscii -intf-suffix .ml -no-alias-deps -open Uspf__ -o lib/.uspf.objs/native/uspf.cmx -c -impl lib/uspf.ml)
- File "lib/uspf.ml", line 683, characters 13-33:
- 683 |           @@ R.failwith_error_msg (Macro.expand_macro ctx macro)) in
-                    ^^^^^^^^^^^^^^^^^^^^
- Error: Unbound value R.failwith_error_msg
```

but I was not 100% sure

Reply:
No `failwith_error_msg` is from `rresult` and was added in rresult.0.6.0. I’ve added the proper constraint.

Reply:
now yes ^^"""
```
#=== ERROR while compiling uspf.0.0.1 =========================================#
# context              2.1.1 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/dinosaure/uspf/releases/download/0.0.1/uspf-0.0.1.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/uspf.0.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p uspf -j 31
# exit-code            1
# env-file             ~/.opam/log/uspf-16422-77ee5a.env
# output-file          ~/.opam/log/uspf-16422-77ee5a.out
### output ###
#       ocamlc lib/.uspf.objs/byte/uspf.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.uspf.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base64/rfc2045 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigarray-overlap -I /home/opam/.opam/4.08/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/coin -I /home/opam/.opam/4.08/lib/colombe -I /home/opam/.opam/4.08/lib/colombe/emile -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/emile -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/hmap -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ke -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mrmime -I /home/opam/.opam/4.08/lib/pecu -I /home/opam/.opam/4.08/lib/prettym -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rosetta -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/unstrctrd -I /home/opam/.opam/4.08/lib/unstrctrd/parser -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/uuuu -I /home/opam/.opam/4.08/lib/yuscii -intf-suffix .ml -no-alias-deps -open Uspf__ -o lib/.uspf.objs/byte/uspf.cmo -c -impl lib/uspf.ml)
# File "lib/uspf.ml", line 1:
# Error: The implementation lib/uspf.ml
#        does not match the interface lib/.uspf.objs/byte/uspf.cmi:
#        ...
#        In module Macro:
#        Values do not match:
#          val expand_string :
#            Uspf__.Map.t ->
#            string -> ([ `raw ] Domain_name.t, Rresult.R.msg) Result.result
#        is not included in
#          val expand_string :
#            ctx ->
#            string -> ([ `raw ] Domain_name.t, [> `Msg of string ]) result
#        File "lib/uspf.mli", line 120, characters 2-95: Expected declaration
#        File "lib/uspf.ml", line 223, characters 6-19: Actual declaration
```